### PR TITLE
Add Auth SDK Docs

### PR DESCRIPTION
**Pull request by TommyJ1994**
### Motivation
Create Documentation for the Auth SDK.

JIRA: [AEROGEAR-2118](https://issues.jboss.org/browse/AEROGEAR-2118)

### Description
Create Documentation for the Auth SDK.

### Progress

- [x] Create Initial Skeleton and Links https://github.com/aerogear/aerogear-ios-sdk/commit/c67704ca611ba6f99641e418438d28a3a012db72
- [x] Adding Dependency Docs https://github.com/aerogear/aerogear-ios-sdk/pull/55/commits/d3d9a9b9a70ebff78e27faf75e89994f5e14a483
- [x] Configuration file Docs https://github.com/aerogear/aerogear-ios-sdk/pull/55/commits/c19d367b32808a3d822b5cf650dc8da6e67edde0
- [x] Initializing the SDK Docs https://github.com/aerogear/aerogear-ios-sdk/pull/55/commits/ca5115f6a72ac836b5cd600a626091b9391a9c91
- [x] Retrieving the current user Docs https://github.com/aerogear/aerogear-ios-sdk/pull/55/commits/d755720e8836e6fae757064597f163d65ed58c6c
- [x] Authenticating Docs https://github.com/aerogear/aerogear-ios-sdk/pull/55/commits/6ac3b15cb44189d2b99e9c53c3aed4162918c2b0
- [x] Retrieving a users roles Docs https://github.com/aerogear/aerogear-ios-sdk/pull/55/commits/da636a486a342ae5680fa1fb20f3a652f1838278
- [x] Logging out Docs https://github.com/aerogear/aerogear-ios-sdk/pull/55/commits/7e9b06073a03cec88f11cae033d543dbe8b78a33

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,13 +14,15 @@ AeroGear Services consist of set of separate SDKs
 
 - [Core](./docs/core/README.adoc)
 Core library used as base for other SDKs
+- [Metrics](./docs/auth/README.adoc)
```
Comment (rachael-oregan) [inline README.md]: [*Auth]

File: docs/auth/README.adoc [hunk 2]
```diff
@@ -1,12 +1,164 @@
...
 
-AeroGear Services auth
+Mobile authentication SDK based on link:http://www.keycloak.org/[Keycloak] using link:http://openid.net/connect/[OpenID Connect].
 
 == Usage
 
-TODO: Add usage
+To use the Auth SDK you'll first need to:
+
+* Have a Keycloak instance.
```
Comment (rachael-oregan) [inline docs/auth/README.adoc]: Should we have a section about setting up Keycloak?

File: docs/auth/README.adoc [hunk 3]
```diff
@@ -1,12 +1,164 @@
...
+    let accessToken = currentUser.accessToken
+    // Get the users identity token
+    let identityToken = currentUser.identityToken
+} catch {
+    fatalError("Error Retrieving User: \(error).")
+}
+----
+
+=== Authenticating
+The authentication flow can be triggered using the `login()` function which provided a callback.
```
Comment (rachael-oregan) [inline docs/auth/README.adoc]: *with a provided callback?

File: docs/auth/README.adoc [hunk 4]
```diff
@@ -1,12 +1,164 @@
...
+
+[source,swift]
+----
+// client roles
+let clientRoles = currentUser.clientRoles
+// realm roles
+let realmRoles = currentUser.realmRoles
+----
+
+In order to check if a user has a specific role you can invoke the `hasClientRole()` and `hasRealmRole() methods and provide the role name to check for.
```
Comment (rachael-oregan) [inline docs/auth/README.adoc]: Missing a ` for hasRealmRole()

**Comment by TommyJ1994** [comment]
@rachael-oregan Thanks for review. Yeah, I left it out originally as i thought it might be better to have it in general 5.x docs as its relevant to all mobile platforms, but I don't think we have anything of this sort yet so I have added in this section for now to this doc, which is the same as the one in the Android auth doc.

File: docs/auth/README.adoc [hunk 5]
```diff
@@ -1,12 +1,184 @@
...
+
+[source,swift]
+----
+// client roles
+let clientRoles = currentUser.clientRoles
+// realm roles
+let realmRoles = currentUser.realmRoles
+----
+
+In order to check if a user has a specific role, you can invoke the `hasClientRole()` method or a `hasRealmRole() methods and provide the role name to check for.
```
Comment (rachael-oregan) [inline docs/auth/README.adoc]: This just needs to change from `hasRealmRole() to `hasRealmRole()`

Comment (TommyJ1994) [reply docs/auth/README.adoc]: Ah, fixed in https://github.com/aerogear/aerogear-ios-sdk/pull/55/commits/763b1b3b16d3dfe6b785b5e1852bd08acae8d1fe

**Pull request merged**